src/tlo/methods/wasting.py

```diff
@@ -1,41 +1,1281 @@
+    individual incident cases to represent onset. It determines those who will progress to severe wasting
+    (WHZ < -3) and schedules the event to update on properties. These are events occurring without the input
+    of interventions, these events reflect the natural history only.
+    """
+    AGE_GROUPS = {0: '0y', 1: '1y', 2: '2y', 3: '3y', 4: '4y'}
+
+    def __init__(self, module):
+        """schedule to run every month
+        :param module: the module that created this event
+        """
+        self.repeat_months = 1
+        super().__init__(module, frequency=DateOffset(months=self.repeat_months))
+        assert isinstance(module, Wasting)
+
+    def apply(self, population):
+        """Apply this event to the population.
+        :param population: the current population
+        """
+        df = population.props
+        rng = self.module.rng
+
+        # # # INCIDENCE OF WASTING # # # # # # # # # # # # # # # # # # # # #
+        # Determine who will be onset with wasting among those who are not currently wasted -------------
+        inc_wasting = df.loc[df.is_alive & (df.age_exact_years < 5) & (
+                df.un_WHZ_category == 'WHZ>=-2')]
+        incidence_of_wasting = self.module.wasting_models.wasting_incidence_lm.predict(inc_wasting,
+                                                                                       rng=rng
+                                                                                       )
+        wasting_idx = inc_wasting.index
+        # update the properties for wasted children
+        df.loc[wasting_idx[incidence_of_wasting], 'un_ever_wasted'] = True
+        df.loc[wasting_idx[incidence_of_wasting], 'un_last_wasting_date_of_onset'] = self.sim.date
+        # start as moderate wasting
+        df.loc[wasting_idx[incidence_of_wasting], 'un_WHZ_category'] = '-3<=WHZ<-2'
+        # start without treatment
+        df.loc[wasting_idx[incidence_of_wasting], 'un_am_treatment_type'] = 'none'
+        # --------------------------------------------------------------------
+        # Add these incident cases to the tracker
+        for person in wasting_idx:
+            wasting_severity = df.at[person, 'un_WHZ_category']
+            age_group = WastingPollingEvent.AGE_GROUPS.get(df.loc[person].age_years, '5+y')
+            # if wasting_severity != 'WHZ>=-2':
+            self.module.wasting_incident_case_tracker[age_group][wasting_severity].append(self.sim.date)
+
+        # ---------------------------------------------------------------------
+
+        # # # PROGRESS TO SEVERE WASTING # # # # # # # # # # # # # # # # # #
+        # Determine those that will progress to severe wasting (WHZ < -3) and schedule progression event ---------
+        progression_sev_wasting = df.loc[df.is_alive & (df.age_exact_years < 5) &
+                                         (df.un_WHZ_category == '-3<=WHZ<-2')]
```

Comment:
Following today discussion with @mnjowe on this matter (which I can confirm from a closer look at the code), the children do not progress immediately from no wasting to severe wasting. First, new moderate wasting cases occur (through incidence). Then among all children with moderate wasting (new and old cases), those who will progress to severe wasting are determined and `ProgressionSevereWastingEvent` is scheduled for them to happen after duration of moderate wasting, and for those who will not progress to severe wasting, `WastingNaturalRecoveryEvent` is scheduled  to happen after duration of moderate wasting.

@mnjowe, As there is no restriction on those who just initiated wasting, wouldn't this lead to the Events being scheduled multiple times? As the old cases will already have scheduled progression or recovery, but they still have moderate wasting, so it will be determined again whether they will progress or recover?